pandas/_libs/tslibs/conversion.pyx

```diff
@@ -203,9 +203,9 @@ cdef class _TSObject:
     #    int64_t value               # numpy dt64
     #    object tzinfo
 
-    property value:
-        def __get__(self):
-            return self.value
+    @property
+    def value(self):
+        return self.value
```

Comment:
I've been kinda curious how this works namespace-wise.  Why isn't this a recursion error?

Reply:
Yeah, now that you mention it, this does seem a bit strange, even the way it was originally written with the legacy syntax.  I tried some similar things in plain Python and they all raised.

That aside, looking further into the code, it doesn't seem like `value` be a property? Or should at least have a setter?  Maybe I'm missing something obvious (not super familiar with this part of the codebase or Cython in general), but it looks like there are places where `value` is being set, e.g.

https://github.com/pandas-dev/pandas/blob/0e168188811677f9de72a6a5b97253e551b6b04a/pandas/_libs/tslibs/conversion.pyx#L250-L256

(and many other examples further below that)

Trying something similar on `master` raises, albeit in plain Python:
```python
In [1]: from pandas._libs.tslibs.conversion import _TSObject

In [2]: obj = _TSObject()

In [3]: obj.value = 10
---------------------------------------------------------------------------
AttributeError: attribute 'value' of 'pandas._libs.tslibs.conversion._TSObject' objects is not writable
```

So I'm a little confused as to how this working in the snippet I posted from the codebase, but again, could just be my relative inexperience with Cython speaking.

Reply:
Are the test failures related to the change in this discussion?

Reply:
@gfyoung :  I believe so, but I'm not 100% sure.

The test failure reads:
```python
  File "pandas/_libs/tslibs/conversion.pyx", line 268, in pandas._libs.tslibs.conversion.convert_to_tsobject (pandas/_libs/tslibs/conversion.c:3771)
    obj.value = ts
AttributeError: can't set attribute
```
which is just a few lines below the snipped from `conversion.pyx` I posted.

Reply:
hmm, maybe it needs a setter as well. I though this was a problem as was trying to access a c-attribute from  a non-cdef, but that's not the case.

Reply:
Okay, so I tried adding a setter, but it still caused a few failures, this time with `TypeError: 'property' object is not callable`.  Looking into it further, it only appears to fail for older Cython versions, specifically 0.23.  Searching around a bit, apparently `@property` syntax wasn't fully supported for older versions like 0.23 (see [here](https://groups.google.com/forum/#!topic/cython-users/eYZDaXdTiis) and [here](https://stackoverflow.com/q/41289669/3339965)).

I've tried a few things, but couldn't get this working without using some type of property syntax.   Would it be best to just revert this to the legacy syntax, possibly with a TODO note to change once we bump the Cython version?  Or is there a non-property way of getting this to work that I've overlooked?

Reply:
Why don't you revert your changes to this `property` syntax so that you have a clean commit that passes all tests.  Then you can mess around with this problematic instance in a separate commit.

Reply:
what version of cython supports this?
not averse to bumping it

Reply:
Looks like 0.24.0.  Was able to reproduce the error locally using 0.23.x, then get it working on 0.24.0.

Reply:
ok would take this as an update then to 0.24
need to update setup.py, some ci files, install.rst and add a more in  whatsnew

Reply:
Sounds good. Will open an issue to have something to reference in the whatsnew, and give this more visibility on the off chance that someone has an objection to the bump.  And here I was thinking that this PR would be an easy and straightforward way to start getting into to the Cython part of the codebase!